RxSwift/Traits/PrimitiveSequence/PrimitiveSequence+Concurrency.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  PrimitiveSequence+Concurrency.swift
+//  RxSwift
+//
+//  Created by Shai Mishali on 22/09/2021.
+//  Copyright © 2021 Krunoslav Zaher. All rights reserved.
+//
+
+import Foundation
+
+#if swift(>=5.5) && canImport(_Concurrency) && !os(Linux)
+@available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
+public extension PrimitiveSequenceType where Trait == SingleTrait {
+    /// Allows awaiting the success or failure of this `Single`
+    /// asynchronously via Swift's concurrency features (`async/await`)
+    ///
+    /// A sample usage would look like so:
+    ///
+    /// ```swift
+    /// do {
+    ///     let value = try await single.value
+    /// } catch {
+    ///     // Handle error
+    /// }
+    /// ```
+    var value: Element {
+        get async throws {
+            try await withCheckedThrowingContinuation { continuation in
+                _ = self.subscribe(
```

Comment:
Shouldn't subscriptions be disposed when tasks gets cancelled? So e.g. this property would look like something like this:
```
var value: Element {
    get async throws {
        var disposable: Disposable?

        return try await withTaskCancellationHandler(
            operation: {
                try await withCheckedThrowingContinuation { continuation in
                    disposable = self.subscribe(
                        onSuccess: { continuation.resume(returning: $0) },
                        onFailure: { continuation.resume(throwing: $0) }
                    )
                }
            },
            onCancel: { [disposable] in
                disposable?.dispose()
            }
        )
    }
}
```
Here's the best piece of documentation that I've found for this topic:
https://github.com/apple/swift-evolution/blob/main/proposals/0304-structured-concurrency.md#cancellation-handlers

Reply:
+1. Boxing might be needed though.
cf. https://gist.github.com/inamiy/0f11034f3dd658ad699a9ac5d2714b2d

Reply:
@inamiy Huh, I thought I've found an easy workaround for the compiler error I was facing, but you're right - `disposable` will always be nil in `onCancel` and boxing should do the trick.

Reply:
Use a single assignment disposable.